File: in_Java/0007 Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        while(x != 0){
            int end = x % 10;
            int newres = res * 10 + end;
            if((newres - end) / 10 != res) return 0;
            res = newres;
            x /= 10;
        }
        return res;
    }
}

File: in_Python/0007 Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        intMax = 2147483647
        intMin = -2147483648
        sign = 1
        if x < 0:
            sign = -1
        x = abs(x)
        n = 0
        while x != 0:
            n = n * 10 + x % 10
            x /= 10
        r = sign * n
        if r > intMax or r < intMin:
            return 0
        else:
            return r